### PR TITLE
#42 - assets controller

### PR DESCRIPTION
**Pull request by MrHDOLEK**
- add assets rout and controller

### CODE REVIEW

File: src/Http/Routing/WebRouting.php [hunk 1]
```diff
@@ -62,5 +63,9 @@ public function wire(): void
             $this->router->put("/speakers/{speaker}", "update")->name("speakers.update");
             $this->router->delete("/speakers/{speaker}", "destroy")->name("speakers.destroy");
         });
+
+        $this->router->controller(AssetsController::class)->group(function (): void {
+            $this->router->get("/static/{folder}/{file}", "index")->name("assets");
```
Comment (krzysztofrewak) [inline src/Http/Routing/WebRouting.php:68]: What about more complicated path structures?

File: src/Http/Controllers/AssetsController.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class AssetsController extends Controller
+{
+    private const ASSETS_PATH = "static";
+
+    public function index(Request $request): Response
+    {
+        $path = resource_path() . "/" . self::ASSETS_PATH . "/" . $request->route("folder") . "/" . $request->route("file");
+        if (!File::exists($path)) {
+            return FacadesResponse::make("", Response::HTTP_NOT_FOUND);
```
Comment (krzysztofrewak) [inline src/Http/Controllers/AssetsController.php:20]: We are not using facades here. Please inject a proper dependency here.

File: src/Http/Controllers/AssetsController.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+namespace Blumilk\Meetup\Core\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Response as FacadesResponse;
+
+class AssetsController extends Controller
+{
+    private const ASSETS_PATH = "static";
```
Comment (krzysztofrewak) [inline src/Http/Controllers/AssetsController.php:14]: Maybe it should be a StaticCotroller? Assets and static words should be unified.

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: src/Http/Routing/WebRouting.php (hunk 1 above)
Comment (krzysztofrewak) [reply src/Http/Routing/WebRouting.php:68]: This should be helpful: https://laravel.com/docs/8.x/routing#parameters-encoded-forward-slashes

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Review by kamilpiech97 — APPROVED** [review]

File: src/FilesFilter.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core;
+
+class FilesFilter
+{
+    protected const FILES = [
```
Comment (kamilpiech97) [inline src/FilesFilter.php:9]: Maybe this `const` should be more readable? `FILES` are a broad concept.

File: src/FilesFilter.php [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core;
+
+class FilesFilter
```
Comment (kamilpiech97) [inline src/FilesFilter.php:7]: Maybe we will create directory for files like that?

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: src/FilesFilter.php (hunk 4 above)
Comment (krzysztofrewak) [reply src/FilesFilter.php:9]: Name `FilesFilter` also should be more precise. Or maybe you could just put this as a function in your controller?

Comment (MrHDOLEK) [reply src/FilesFilter.php]: I think IGNORE FILES is a clearer name

File: src/Http/Controllers/StaticController.php [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+namespace Blumilk\Meetup\Core\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\File;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+
+class StaticController extends Controller
+{
+    protected const IGNORE_FILES = [
```
Comment (krzysztofrewak) [inline src/Http/Controllers/StaticController.php:14]:
```suggestion
    protected const IGNORED_FILES = [
```

Comment (krzysztofrewak) [reply src/Http/Controllers/StaticController.php:14]: And why one const has protected and second one - private accessor?

File: src/Http/Controllers/StaticController.php [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        $headers = [
+            "Content-Type" => File::mimeType($path),
+        ];
+
+        return response()->file($path, $headers);
+    }
+
+    protected function filesFilter(string $fileName): bool
+    {
+        return in_array($fileName, self::IGNORE_FILES, true);
```
Comment (krzysztofrewak) [inline src/Http/Controllers/StaticController.php:35]:
```suggestion
        return in_array($fileName, static::IGNORE_FILES, true);
```

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Review by MichalMyskow — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**